hail/python/hail/conftest.py

```diff
@@ -4,11 +4,23 @@
 import hail as hl
 import hail.expr.aggregators as agg
 
+import doctest
+
+NOTEST = doctest.register_optionflag('NOTEST')
+
 
 @pytest.fixture(autouse=True)
-def always_true(monkeypatch):
+def patch_doctest_check_output(monkeypatch):
     # FIXME: remove once test output matches docs
-    monkeypatch.setattr('doctest.OutputChecker.check_output', lambda a, b, c, d: True)
+    base_check_output = doctest.OutputChecker.check_output
+
+    def patched_check_output(self, want, got, optionflags):
+        return ((not want)
+                or (want.strip() == 'None')
```

Comment:
I'm not sure this is what we want. IMO, we should change our examples to be `blah is None`. With this setting, if something is supposed to return `None` but actually returns `[]` it will not fail, right?

Reply:
ah, good point.

Reply:
want and got are strings, so `not want` will only return True if want is `''`

Reply:
I have again convinced myself this is right

Reply:
If `got == '[]'` and `want == 'None'`, does this code say "ignore this test"?

It seems like the Doctest Approved Way™️ of doing this is `blah is None`. I'm rather OK with that honestly since it's less surprising to the user who copy-pastes a command into Python's repl. Regardless, we can do a follow up doctest cleanup.

Reply:
ahhh, I see. you are right -- we're allowing anything through if want is `'None'`. We should be checking that the result is either `'None'` (which should be impossible?) or `<BLANKLINE>` (which I believe is the doctest way of doing it)